Source/BugsnagConfiguration.m

```diff
@@ -46,43 +49,60 @@ @interface BugsnagNotifier ()
 @end
 
 @interface BugsnagConfiguration ()
+
 @property(nonatomic, readwrite, strong) NSMutableArray *beforeNotifyHooks;
 @property(nonatomic, readwrite, strong) NSMutableArray *beforeSendBlocks;
 @property(nonatomic, readwrite, strong) NSMutableArray *beforeSendSessionBlocks;
 @end
 
 @implementation BugsnagConfiguration
 
-- (id)init {
-    if (self = [super init]) {
-        _metaData = [[BugsnagMetaData alloc] init];
-        _config = [[BugsnagMetaData alloc] init];
-        _apiKey = @"";
-        _sessionURL = [NSURL URLWithString:@"https://sessions.bugsnag.com"];
-        _autoDetectErrors = YES;
-        _notifyURL = [NSURL URLWithString:BSGDefaultNotifyUrl];
-        _beforeNotifyHooks = [NSMutableArray new];
-        _beforeSendBlocks = [NSMutableArray new];
-        _beforeSendSessionBlocks = [NSMutableArray new];
-        _notifyReleaseStages = nil;
-        _breadcrumbs = [BugsnagBreadcrumbs new];
-        _automaticallyCollectBreadcrumbs = YES;
-        _autoTrackSessions = YES;
-#if !DEBUG
+/**
+ * Should not be called, but if it _is_ then fail meaningfully rather than silently
+ */
+- (instancetype)init {
+    @throw BSGInitError;
+}
+
+/**
+ * The designated initializer.
+ */
+-(instancetype)initWithApiKey:(NSString *)apiKey {
+    if (! [self isValidApiKey:apiKey]) {
+        @throw BSGApiKeyError;
```

Comment:
I played around with this a bit, and noticed its not catchable in any way from Swift. The primary case where this might be missed in development is that sometimes people only initialize  bugsnag for their beta/release builds, so this could be an unexpected crash in a release.

I thought of an alternative here, where `initWithApiKey` takes an error parameter and converts into a `try` statement in Swift. Something like:

```objc
// BugsnagConfiguration.h
- (instancetype _Nullable)initWithApiKey:(NSString *_Nonnull) apiKey error:(NSError *_Nullable *_Nullable)error
    NS_DESIGNATED_INITIALIZER NS_SWIFT_NAME(init(_:)) __attribute__((swift_error(nonnull_error)));
```

which then makes the Swift initializer testable for errors using

```swift
let config = try BugsnagConfiguration("{a totally valid API key here}")
```

Does this make sense and seem like a workable way forward?

Reply:
Yes, good spot.  At the acknowledged (idiomatic) expense of a slightly longer invocation.  These changes - with the additional Swift tests they make possible - are now checked in.